pkg/controllers/workgenerator/controller_integration_test.go

```diff
@@ -426,33 +429,67 @@ var _ = Describe("Test Work Generator Controller", func() {
-					verifyBindStatusNotAppliedWithFailedPlacement(binding, false) // mark the work applied but still not available with failed manifests
+					verifyBindStatusNotAppliedWithTwoPlacements(binding, false, true, false, false) // mark the work applied but still not available with failed manifests
 					// change one of the failed manifests to be applied
-					markWorkAsAppliedButNotAvailableWithFailedManifest(&work)
+					markWorkAsAppliedButNotAvailableWithFailedManifest(&work, false)
 					// check the binding status that it should be applied but not available and with two failed placement
-					verifyBindStatusNotAvailableWithTwoFailedPlacement(binding, false)
+					verifyBindStatusNotAvailableWithTwoPlacements(binding, false, true, false)
 					// mark one of the failed manifests as available but no change in the overall condition
-					markOneManifestAvailable(&work)
+					markOneManifestAvailable(&work, false)
 					// check the binding status that it should be applied but not available and with one failed placement
-					verifyBindStatusNotAvailableWithOneFailedPlacement(binding, false)
+					verifyBindStatusNotAvailableWithOnePlacement(binding, false, true)
 					// mark the work available directly
 					markWorkAvailable(&work)
 					// check the binding status that it should be marked as available true eventually
-					verifyBindStatusAvail(binding, false)
+					verifyBindStatusAvail(binding, false, false)
+				})
+
+				It("Should update the binding when the diffed placement list has changed but over all condition didn't change", func() {
+					// mark the work as not applied with failed manifests and diffed manifests
+					markWorkWithFailedToApplyAndNotAvailable(&work, true, false)
+					// check the binding status that it have failed placements and diffed placements
+					verifyBindStatusNotAppliedWithTwoPlacements(binding, false, true, true, false)
+					// change one of the failed manifests to be applied and diff manifests to be aligned (applied is now true so no diff placements)
+					markWorkAsAppliedButNotAvailableWithFailedManifest(&work, false)
+					// check the binding status that it should be applied but not available and with two failed placement
+					verifyBindStatusNotAvailableWithTwoPlacements(binding, false, true, false)
+					// mark one of the failed manifests as available but no change in the overall condition
+					markOneManifestAvailable(&work, false)
+					// check the binding status that it should be applied but not available and with one failed placement
+					verifyBindStatusNotAvailableWithOnePlacement(binding, false, true)
+					// mark the work available directly
+					markWorkAvailable(&work)
+					// check the binding status that it should be marked as available true eventually
+					verifyBindStatusAvail(binding, false, false)
+				})
+
+				It("Should update the binding when the drifted placement list has changed but over all condition didn't change", func() {
```

Comment:
I wonder which step tested the "drifted placement list"?

Reply:
Anytime we verify the binding status. The main purpose of this PR is to make sure that the placements list are copied over from the work to the binding. In this case, since drifted can happen when the `Applied` condition is true or false, I tested to make sure the placement list stays even when it fails and is available.

Reply:
Oh I see the it actually does not change the list. Updated test to have the list actually change.